ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/rules/WrappingRuleTest.kt

```diff
@@ -1871,6 +1871,303 @@ internal class WrappingRuleTest {
+        val code =
+            """
+            fun f1() {
+                val a = 3; val b = 2
+                val key: String = ""; val lambda: () -> Unit = {  }
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            fun f1() {
+                val a = 3;
+                val b = 2
+                val key: String = "";
+                val lambda: () -> Unit = {  }
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(2, 15, "Missing newline after \";\""),
+                LintViolation(3, 26, "Missing newline after \";\""),
+            )
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated statements then wrap each expression to a new line`() {
+        val code =
+            """
+            public fun f2() {
+                // no-op
+            }; public fun f3() {
+                // no-op
+            }
+
+            public fun f4() = 1; public fun f5() {
+                // no-op
+            }
+
+            public class A {
+
+            }; public class B {
+
+            }
+
+            public class C; public class D
+
+            class E {
+                init {
+                    f1(); f2()
+                }
+            }
+
+            enum class INDEX {
+                A, B, C; fun test() = 0
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            public fun f2() {
+                // no-op
+            };
+            public fun f3() {
+                // no-op
+            }
+
+            public fun f4() = 1;
+            public fun f5() {
+                // no-op
+            }
+
+            public class A {
+
+            };
+            public class B {
+
+            }
+
+            public class C;
+            public class D
+
+            class E {
+                init {
+                    f1();
+                    f2()
+                }
+            }
+
+            enum class INDEX {
+                A, B, C;
+                fun test() = 0
+            }
+            """.trimIndent()
+        wrappingRuleAssertThat(code)
+            .hasLintViolations(
+                LintViolation(3, 3, "Missing newline after \";\""),
+                LintViolation(7, 21, "Missing newline after \";\""),
+                LintViolation(13, 3, "Missing newline after \";\""),
+                LintViolation(17, 16, "Missing newline after \";\""),
+                LintViolation(21, 14, "Missing newline after \";\""),
+                LintViolation(26, 13, "Missing newline after \";\""),
+            )
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Issue 1078 - Given a multiline semi separated control flow then wrap each expression to a new line`() {
+        val code =
+            """
```

Comment:
Please split in multiple tests. If needed use a nested test class to group related tests.